File: datalake/src/main/java/com/sequenceiq/datalake/controller/util/UtilController.java
package com.sequenceiq.datalake.controller.util;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Controller;

import com.sequenceiq.common.api.util.versionchecker.ClientVersionUtil;
import com.sequenceiq.common.api.util.versionchecker.VersionCheckResult;
import com.sequenceiq.sdx.api.endpoint.UtilEndpoint;

@Controller
public class UtilController implements UtilEndpoint {

    @Value("${info.app.version:}")
    private String sdxVersion;

    @Override
    public VersionCheckResult checkClientVersion(String version) {
        return ClientVersionUtil.checkClientVersion(sdxVersion, version);
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/UtilController.kt
package com.sequenceiq.cloudbreak.controller

import javax.validation.Valid

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

import com.sequenceiq.cloudbreak.api.endpoint.UtilEndpoint
import com.sequenceiq.cloudbreak.api.model.RDSConfigJson
import com.sequenceiq.cloudbreak.api.model.RdsTestResult
import com.sequenceiq.cloudbreak.controller.validation.rds.RdsConnectionValidator

@Component
class UtilController : UtilEndpoint {

    @Autowired
    private val rdsConnectionValidator: RdsConnectionValidator? = null

    override fun testRdsConnection(@Valid rdsConfigJson: RDSConfigJson): RdsTestResult {
        val rdsTestResult = RdsTestResult()
        try {
            rdsConnectionValidator!!.validateRdsConnection(rdsConfigJson)
            rdsTestResult.connectionResult = "connected"
        } catch (e: BadRequestException) {
            rdsTestResult.connectionResult = e.message
        }

        return rdsTestResult
    }
}
